### PR TITLE
Remove primary navigation component

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
- Remove primary navigation component
- Remove documentation for primary navigation component, but keep page with a link to Service navigation on GOV.UK Design System website
- Remove examples for the primary navigation component
- Update remaining documentation and examples to use Service navigation component, where relevant
- Bump package to v4.0.0, a breaking MAJOR release

Screenshot of retirement notice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "@11ty/eleventy": "^2.0.1",
     "@rollup/plugin-commonjs": "^26.0.0",
     "@rollup/plugin-node-resolve": "^15.0.1",
-    "@x-govuk/govuk-eleventy-plugin": "^6.0.0",
+    "@x-govuk/govuk-eleventy-plugin": "x-govuk/govuk-eleventy-plugin#service-navigation",
```
Comment (paulrobertlloyd) [inline package.json:54]: Depends on https://github.com/x-govuk/govuk-eleventy-plugin/pull/316

**Review by frankieroberto — COMMENTED** [review]
I wonder if we should do a slightly less aggressive deprecation strategy, by keeping the page in the nav, and keeping the previous examples and documentation, but having the "this has been retired" banner at the top, and making it clear that the examples below will only work in versions below 4.0.0?

Could even include a section explaining how to switch from `xGovukPrimaryNavigation({ .. })` to `govukServiceNavigation({ .. })`?

Feels like there might still be a need to keep the documentation around for `xGovukPrimaryNavigation` for a while longer (6 months?) to give people a chance to upgrade.

**Comment by frankieroberto** [comment]
Looks like around 20 public repos have used this component! https://github.com/search?q=xGovukPrimaryNavigation&type=code

**Comment by paulrobertlloyd** [comment]
Why do you spoil all my fun @frankieroberto 🥹

So maybe we do a MINOR release, and add deprecation notices where we can. Am thinking:

- On the docs, add the above banner
- In SCSS, output deprecation warning to the console
- In the Prototype Kit templates, includes a deprecation message in the meta data somewhere
- Possibly in the Nunjucks somehow, include a filter that outputs a deprecation message to the console

Then, in version 4, remove the component as shown in this PR. I might try and achieve the above in a separate PR.

**Comment by frankieroberto** [comment]
@paulrobertlloyd that sounds reasonable to me!

We could also do a bit of proactive outreach to the repos who’re using the component to warn them about the deprecation.

**Pull request closed without merging**